qiskit/compiler/transpiler.py

```diff
@@ -40,6 +50,17 @@
 
 _CircuitT = TypeVar("_CircuitT", bound=Union[QuantumCircuit, List[QuantumCircuit]])
 
+CUSTOM_MAPPING = {
+    "if_else": IfElseOp,
+    "for_loop": ForLoopOp,
+    "while_loop": WhileLoopOp,
+    "switch_case": SwitchCaseOp,
+    "continue": ContinueLoopOp,
+    "break": BreakLoopOp,
+    "unitary": UnitaryGate,
+    "qsd2q": UnitaryGate,
```

Comment:
Where was this coming up? I'm assuming there was a quantum shannon decomposition test where we passed `qsd2q` to test the intermediate synthesis via `transpile()`?

This actually raises an interesting backwards compatibility question for me. Do we have any tests for the edge case someone does something like:

```python

qc = QuantumCircuit(4)
qc.cx(0, 1)
qc.cz(0, 2)
qc.append(MyCustomGate(), [0, 1, 2, 3])
transpile(qc, basis_gates=["cz", "u", "my_custom_gate"])
```

where `my_custom_gate` is the `name` attribute of `MyCustomGate`? I feel like this is a breaking change caused by this PR because we have no mechanism to do this without providing the custom mapping which we don't have an API for. I'm wondering if we want a fallback for when `Target.from_configuration()` fails with the error because there is no mapping for a gate name to handle this.

Reply:
Ah ok that makes sense. That wasn't clear to me at first, because this would also potentially be the case in the absence of pulse gates too. Well if we've got handling for it now then there is nothing to worry about on that front. But we should remove this dictionary if it's not being used anymore.